src/main/java/edu/tamu/eider/app/model/processor/EntityCollectionProcessor.java

```diff
@@ -15,14 +15,15 @@
 
     @Override
     public CollectionModel<Entity> process(CollectionModel<Entity> model) {
-        String baseUrl = ServletUriComponentsBuilder.fromCurrentContextPath().build().toUriString();
-        
-        model.add(
-            of(baseUrl + "/entity/name?name={name}").withRel(LinkRelation.of("findByName"))
-        );
-        model.add(
-            of(baseUrl + "/entity/url?url={url}").withRel(LinkRelation.of("findByUrl"))
-        );
+        if (model.getLinks().getLink("self").toString().contains("entity")) {
```

Comment:
This seems more like a work around. I do not think representation model processor's are used for collections. I believe you will need something like following.
```
public class EntityRepresentationModel extends EntityModel<Entity>

@Component
public classEntityRepresentationModel Assembler  extends RepresentationModelAssemblerSupport<Entity, EntityRepresentationModel>

@Component
public class EntityProcessor implements RepresentationModelProcessor<EntityRepresentationModel>
```

You then can override [toCollectionModel ](https://docs.spring.io/spring-hateoas/docs/1.0.0.M1/apidocs/org/springframework/hateoas/server/mvc/RepresentationModelAssemblerSupport.html#toCollectionModel-java.lang.Iterable-).

Reply:
With further investigation Spring Boot does not have desired affordance for adding links to a specific CollectionModel response. Additionally, the RepresentationModelAssemblerSupport only is used for custom rest resource methods and does not effect the built in PersistentEntityResourceAssembler. The RepresentationModelProcessor does support typed EntityModel but not typed CollectionModel, hence this workaround. Until a better way to do this is afforded this conditional will have to be used.